File: VB/Modules.Contact/ContactsModule.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Microsoft.Practices.Composite.Regions
Imports Microsoft.Practices.Composite.Modularity

Namespace Modules.Contacts
	Public Class ContactsModule
		Implements IModule
		Private ReadOnly regionManager As IRegionManager
		Public Sub New(ByVal regionManager As IRegionManager)
			Me.regionManager = regionManager
		End Sub
        Public Sub Initialize() Implements IModule.Initialize
            regionManager.RegisterViewWithRegion("ContactsRegion", GetType(ContactsView))
        End Sub
	End Class
End Namespace

File: CS/Modules.Contact/ContactsModule.cs
using Prism.Modularity;
using Prism.Regions;

namespace Modules.Contacts {
    public class ContactsModule : IModule {
        private readonly IRegionManager regionManager;
        public ContactsModule(IRegionManager regionManager) {
            this.regionManager = regionManager;
        }
        void IModule.Initialize() {
            regionManager.RegisterViewWithRegion("ContactsRegion", typeof(ContactsView));
        }
    }
}
